src/kernel.ts

```diff
@@ -884,12 +905,15 @@ class Kernel implements IKernel {
       return;
     }
     var content = msg.content as ICommOpen;
-    if (!content.target_module) {
-      this.commOpened.emit(msg);
+    this.commOpened.emit(msg);
+    var targetName = content.target_name;
+    var moduleName = content.target_module;
+    if (this._targetRegistry[targetName]) {
+      this._targetRegistry[targetName](msg);
+    }
```

Comment:
The widgets has a few registries scattered about, and all of the registries take precedence over the differed loading mechanism.  Not to say we have to do that here, but my reasoning was it made it easy for the user to come along and override something internal, during live exploration in the notebook.